components/enrichers/reachability/internal/atom/purl/purl.go

```diff
@@ -0,0 +1,143 @@
+package purl
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// Parser allows to extract information from purls - https://github.com/package-url/purl-spec.
+type Parser struct {
+	matcherPurlPkg             *regexp.Regexp
+	matcherPurlTrailingVersion *regexp.Regexp
+	matcherPurlVersion         *regexp.Regexp
+}
+
+func NewParser() (*Parser, error) {
+	purlPkg, err := regexp.Compile(`(?P<p1>[^/:]+/(?P<p2>[^/]+))(?:(?:.|/)v\d+)?@`)
```

Comment:
why are we using regexp to parse this and not https://github.com/package-url/packageurl-go?

Reply:
Will look into it!